sdk/core/core/inc/az_span.h

```diff
@@ -84,30 +88,37 @@ AZ_NODISCARD AZ_INLINE int32_t az_span_capacity(az_span span) { return span._int
   }
 
 /**
- * @brief Creates a az_span expression. For example,
- * `some_function(AZ_SPAN_FROM_STR("Hello world"));`
- * where
- * `void some_function(const az_span span);`
+ * @brief The AZ_SPAN_FROM_STR macro returns an az_span expression over a literal string. For
+ * example: `some_function(AZ_SPAN_FROM_STR("Hello world"));` where `void some_function(const
+ * az_span span);`
  */
 #define AZ_SPAN_FROM_STR(STRING_LITERAL) (az_span) AZ_SPAN_LITERAL_FROM_STR(STRING_LITERAL)
 
 /**
- * @brief Init span
+ * @brief az_span_init returns a span over a byte buffer.
  *
+ * @param[in] ptr The memory address of the 1st byte in the byte buffer
+ * @param[in] length The number of bytes initialized in the byte buffer
+ * @param[in] capacity The number of total bytes in the byte buffer
+ * @return az_span The az_span "view" over the byte buffer
  */
 AZ_NODISCARD az_span az_span_init(uint8_t* ptr, int32_t length, int32_t capacity);
 
 /**
- * @brief Creates a span from zero terminated string
+ * @brief az_span_from_str returns an az_span from a 0-terminated array of bytes (chars)
```

Comment:
A span is a byte buffer. Bytes are 0-255. However, you can reinterpret bytes however you want. So, you can "cast" a byte to a char which is really a signed 8-bit integer. An unsigned char is an unsigned 8-bit integer which is the same as a byte.

All buffers are 0-terminated: You'll eventually hit a 0 or end of the address space and then wrap around and eventually hit a 0.

Reply:
> An unsigned char is an unsigned 8-bit integer which is the same as a byte.

Great, so maybe update the comment to explicitly state `(unsigned chars)` instead of just `(chars)`.

```suggestion
 * @brief az_span_from_str returns an az_span from a 0-terminated array of bytes (unsigned chars)
```

> You'll eventually hit a 0 or end of the address space and then wrap around and eventually hit a 0.

But could that happen after more than int32 max bytes? And if so, does our length count potentially overflow to negative?

Reply:
It is idiomatic in C to have a char* point to a string; not an unsigned char*. So I don't want to make this change in docs or in the function signature. 

Our documented contract is that we accept a pointer to a C string; which, by definition means a 0-terminated string. If the customer violates the contract, then behavior is unpredictable. So, it doesn't matter if the "buffer" (which is not what our contract says it expects) is not 0-terminated.

Reply:
Sounds good. Thanks for providing such details and help talk through the design and doc nuances :)